### PR TITLE
Update references to previous repository + bump doc to 0.3 version

### PR DESCRIPTION
**Pull request by danielezonca**
PR to update references to previous repository

### CODE REVIEW

**Comment by danielezonca** [comment]
@jackdelahunt can you please have a look?

File: examples/provider/main.tf [hunk 1]
```diff
@@ -1,7 +1,8 @@
 terraform {
   required_providers {
     rhoas = {
-      source  = "pmuir/rhoas"
+      source  = "registry.terraform.io/redhat-developer/rhoas"
+      version = "0.1"
```
Comment (jackdelahunt) [inline examples/provider/main.tf]: We are currently on `0.3.0`

Comment (jackdelahunt) [reply examples/provider/main.tf]: Maybe you can change it in the `docs/index.md` as that is out of date aswell

Comment (danielezonca) [reply examples/provider/main.tf]: Sure, updated all usages of `"0.1"` to `"0.3"` 👍 
It should be nice with another PR to automate this replace as part of the release process :)

**Comment by danielezonca** [comment]
@jackdelahunt 
Can you please approve the workflow so CI can run (even if I don't expect this PR to have impact on behaviour, it is mainly doc)? 
image

**Comment by jackdelahunt** [comment]
Failed CI is expected LGTM 👍

**Pull request merged**